compiler-rt/lib/nsan/nsan_interceptors.cpp

```diff
@@ -93,8 +93,10 @@ INTERCEPTOR(char *, strfry, char *s) {
 }
 
 INTERCEPTOR(char *, strsep, char **Stringp, const char *delim) {
```

Comment:
why do you need string function for nsan?

Reply:
they modify memory, so by intercepting them nsan modifies the shadow memory accordingly.
It rarely matters in practice though (based on what i've seen so far), though one can come up with a contrived / artificial example.